src/features/app/components/Navbar/Menu.test.js

```diff
@@ -0,0 +1,91 @@
+import '@testing-library/jest-dom/extend-expect';
+
+import { fireEvent, renderWithRouter, waitFor } from 'testUtils';
+import {
+  userData,
+  authenticationHeaders,
+  mockSignOutSuccess,
+} from 'testUtils/mocks/auth';
+import Navbar from './Navbar';
+
+describe('Menu', () => {
+  const fakeState = {
+    auth: {
+      user: userData,
+      session: authenticationHeaders,
+    },
+  };
+
+  it('navigates to settings', async () => {
+    const { history, getByText, getByRole } = renderWithRouter(<Navbar />, {
+      history: ['/'],
+      state: fakeState,
+    });
+
+    const menuButton = getByRole('button');
+    fireEvent.click(menuButton);
+
+    const settingsButton = getByText('Account settings');
+    fireEvent.click(settingsButton);
+
+    await waitFor(() => {
+      expect(settingsButton).toBeInTheDocument();
+      expect(history.location.pathname).toEqual('/settings');
+    });
+  });
+
+  it('logs out', async () => {
+    const mockedRequest = mockSignOutSuccess();
+
+    const { history, getByText, getByRole } = renderWithRouter(<Navbar />, {
+      history: ['/'],
+      state: fakeState,
+    });
+
+    const menuButton = getByRole('button');
+    fireEvent.click(menuButton);
+
+    const signOutButton = getByText('Sign out');
+    fireEvent.click(signOutButton);
+
+    await waitFor(() => {
+      expect(mockedRequest.isDone()).toBeTruthy();
+      expect(signOutButton).toBeInTheDocument();
+      expect(history.location.pathname).toEqual('/sign-in');
+    });
+  });
+
+  it('navigates to sign in', async () => {
+    const { history, getAllByText, getByRole } = renderWithRouter(<Navbar />, {
+      history: ['/forgot-password'],
+    });
+
+    const menuButton = getByRole('button');
+    fireEvent.click(menuButton);
+
+    const signInButton = getAllByText('Sign in')[1];
```

Comment:
Why you did this in this way? I think that it could be "hard" to maintain because is like an implementation detail.
Why not `getByRole('button', { name: 'Sign in' });` ?